spi_nand_flash/src/nand.c

```diff
@@ -279,18 +285,23 @@ esp_err_t spi_nand_flash_read_sector(spi_nand_flash_device_t *handle, uint8_t *b
     dhara_error_t err;
     esp_err_t ret = ESP_OK;
 
+    uint8_t *buf = heap_caps_malloc(handle->page_size, MALLOC_CAP_DMA | MALLOC_CAP_8BIT);
```

Comment:
Can we allocate the read buffer in `spi_nand_flash_init_device`, so that we don't have to re-allocate it each time?